js/interface.js

```diff
@@ -287,14 +287,16 @@ function initLinkProvider(item) {
 
 function initListener(provider, item) { 
   window.addEventListener('message', function onMessage(event) {
-    if (event.data === 'cancel-button-pressed') {
-      switch (provider) {
-        case 'icon':
-          onIconClose(item);         
-          break;
-        case 'image':
-          onImageClose(item);
-          break;
+    if (event.data === 'cancel-button-pressed' || event.data === 'save-widget') {
```

Comment:
This code looks weird.

```
if (A || B) {
  if (A) {
    // ...
  }
}
```

what is the point of having this?

Reply:
@squallstar 
This was needed to removeListener for cancel button when we are pressing save & close button.
But I think I can do better code than this, and I'll remake this part.

Reply:
@squallstar Changed weird code.